src/web/transactions.rs

```diff
@@ -3,122 +3,103 @@ use std::collections::BTreeSet;
 use crate::{iroha_client_wrap::QueryBuilder, web::etc::HashDeser};
 
 use super::{
-    etc::{SerScaleHex, SignatureDTO, Timestamp},
-    get, web, AppData, Paginated, PaginationQueryParams, Scope, WebError,
+    etc::{SerScaleHex, Timestamp},
+    get,
+    web,
+    AppData,
+    Paginated,
+    PaginationQueryParams,
+    Scope,
+    WebError,
 };
+use crate::web::etc::SignatureDTO;
 use color_eyre::{eyre::Context, Result};
-use iroha_core::tx::{
-    Executable, RejectedTransaction, TransactionRejectionReason, TransactionValue, Txn,
-};
-use iroha_crypto::{Hash, HashOf, Signature};
+use iroha_core::tx::{Executable, TransactionValue, VersionedSignedTransaction};
+use iroha_crypto::{HashOf, Signature, SignaturesOf};
+use iroha_data_model::block::CommittedBlock;
 use iroha_data_model::prelude::{
-    FindAllTransactions, FindTransactionByHash, Instruction, Payload, SignedTransaction, UnlimitedMetadata,
+    FindAllTransactions, FindTransactionByHash, InstructionBox, TransactionQueryResult,
+    UnlimitedMetadata,
+};
+use iroha_data_model::transaction::{
+    error::model::TransactionRejectionReason, model::TransactionPayload,
 };
+
+use core::num::{NonZeroU32, NonZeroU64};
 use serde::Serialize;
 
 #[derive(Serialize)]
-#[serde(tag = "t", content = "c")]
-pub enum TransactionDTO {
-    Committed(CommittedTransactionDTO),
-    Rejected(RejectedTransactionDTO),
+pub struct TransactionDTO {
+    hash: SerScaleHex<HashOf<VersionedSignedTransaction>>,
+    block_hash: SerScaleHex<HashOf<CommittedBlock>>,
+    payload: TransactionPayloadDTO,
+    signatures: BTreeSet<SignatureDTO>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    rejection_reason: Option<SerScaleHex<TransactionRejectionReason>>,
```

Comment:
**TODO:** check what exactly `skip_serializing_if` does: `key: null`, `key: undefined`, or no `key` at all in the serialized structure?

Reply:
Clarification : 
As rejection_reason is an ```Option Enum``` by default so it would either have a None return type or Some< T> . 
So are we discussing what if this Some< T> is ```null``` or ```undefined``` ?

Because if ```error.clone()``` is None, it indicates that the transaction was not rejected and therefore there is no rejection reason available. As a result, the rejection_reason field in the serialized JSON output will be skipped due to the ```skip_serializing_if attribute```.

Reply:
Okay. Thus, the conclusion is:

```ts
interface TransationDTO {
  // ...
  rejection_reason?: string
}
```